packages/shared/src/components/PostsSearch.tsx

```diff
@@ -105,9 +107,11 @@ export default function PostsSearch({
     }
   };
 
-  useEffect(() => {
-    searchBoxRef.current?.querySelector('input').focus();
-  }, [searchBoxRef]);
+  if (autoFocus) {
+    useEffect(() => {
+      searchBoxRef.current?.querySelector('input').focus();
+    }, [searchBoxRef]);
+  }
```

Comment:
Not sure conditional hooks are allowed in React.
It's better to put the condition inside the `useEffect`. I know it can mess their internals